### PR TITLE
helm/cluster-operator: make deployment compatible with K8s 1.16

### PR DESCRIPTION
**Pull request by mig4**
Use apps/v1 for Deployment and add a selector to match by app and
version. `extensions/v1beta1` is no longer available in 1.16.

Towards giantswarm/giantswarm#7675.

### CODE REVIEW

**Comment by mig4** [comment]
@rossf7 this is a follow-up to your comment on #797

**Comment by tfussell** [comment]
This is needed for Kubernetes 1.16, so definitely a good change. Can you explain how this helps solve the linked issue though?

**Comment by rossf7** [comment]
> This is needed for Kubernetes 1.16, so definitely a good change. Can you explain how this helps solve the linked issue though?

@tfussell  Yes I think that's the wrong issue. It's an unrelated PM that I'm working on.

@mig4 Could you check that?

**Comment by kopiczko** [comment]
@tfussell this is to address @rossf7 review comment https://github.com/giantswarm/cluster-operator/pull/797#discussion_r350333551

File: helm/cluster-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ metadata:
     version: {{ .Values.project.version }}
 spec:
   replicas: 1
+  selector:
+    matchLabels:
+      app: {{ .Values.project.name }}
+      version: {{ .Values.project.version }}
```
Comment (kopiczko) [inline helm/cluster-operator/templates/deployment.yaml]: This is unrelated considering the PR title. I'd change it to something like `make deployment compatible with K8s 1.16` or something.

**Comment by mig4** [comment]
Sure, I just pretty much copied the commit message from @rossf7's commit in aws-operator. Is there an issue tracking k8s 1.16 I could reference here instead?

**Comment by sslavic** [comment]
There is https://github.com/giantswarm/giantswarm/issues/7428 for upgrading of the server side to 1.16

Prerequisite for 1.16 server side support is https://github.com/giantswarm/giantswarm/issues/7675 - it's tracking tasks for upgrading of the client side to 1.16 and just linked this PR in there.

Thanks for doing this migration part - PR LGTM!

File: helm/cluster-operator/templates/deployment.yaml (hunk 1 above)
Comment (sslavic) [reply helm/cluster-operator/templates/deployment.yaml]: Title adjusted, hope it's more clear. PTAL

Comment (mig4) [reply helm/cluster-operator/templates/deployment.yaml]: Sorry @sslavic didn't notice you changed it and applied my change from the commit, I don't see a way to revert the title, but hope this is OK.

Comment (sslavic) [reply helm/cluster-operator/templates/deployment.yaml]: All good, your change is more expressive/detailed which I like.

Many ACKs - feel free to ship it!
